lib/std/fs/test.zig

```diff
@@ -100,6 +100,25 @@ test "create file, lock and read from multiple process at once" {
     };
 }
 
+test "open file with exclusive nonblocking lock twice (absolute paths)" {
+    const allocator = std.heap.page_allocator;
+
+    const file_paths: [1][]const u8 = .{"zig-test-absolute-paths.txt"};
+    const filename = try fs.path.resolve(allocator, &file_paths);
+    defer allocator.free(filename);
+
+    const file1 = try fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
+    defer file1.close();
+
+    const file2 = fs.createFileAbsolute(filename, .{ .lock = .Exclusive, .lock_nonblocking = true });
+    std.debug.assert(std.meta.eql(file2, error.WouldBlock));
+
+    fs.deleteFileAbsolute(filename) catch |err| switch (err) {
+        error.FileNotFound => {},
```

Comment:
You should probably close `file1`, some systems may not want to unlink open files.
And `FileNotFound` should not be ignored, we _know_ the file exists!

Reply:
I thought the same, but I copied that from [this test](https://github.com/ziglang/zig/blob/6c907a350989aa881cc0b0f71ec81d1ae1b449b7/lib/std/fs/test.zig#L13) which also handles it that way, so I thought there might be a good reason for this.

Reply:
I just tested it: When closing fil1 before opening file2, the WouldBlock doesn't occur.

Reply:
> I just tested it: When closing fil1 before opening file2, the WouldBlock doesn't occur.

Eh? Of course it doesn't, the lock is gone when you close the file.
I suggested to close the file before deleting the file.